### DISCUSSION TITLE
Question on Profile Deployment

### DISCUSSION THREAD

**Discussion by sameerseekala1** [scolladon/sfdx-git-delta#788] [General]
How will 'sgd:source:delta' command work on profiles? If we make 1 FLS change on the profile file with it generate delta  with only that specific change or it shows entire profile file? I tried to test it but every time it shows all the page layout assignments along with the FLS change in the profile.
Please let me know how the command works for profiles TIA.

**Comment by scolladon** [comment]
Hi @sameerseekala1 !

You are not alone facing this particular behavior of the plugin
Here is an issue logged a few days before you [here](https://github.com/scolladon/sfdx-git-delta/issues/787)

I tried to answer there, feel free to raise any remaining questions

**Reply by sameerseekala1** [reply]
Thanks for the response @scolladon.

So that means profile file will have some other items along with the FLS change right? Apart from layouts & IP's what other items delta may include by default?

**Reply by scolladon** [reply]
Here is the list of attributes the plugin does not trim in profile metadata (they will always be present): 
- [custom](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_profile.htm#:~:text=profile%20can%20execute.-,custom,-boolean)
- [description](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_profile.htm#:~:text=47.0%20and%20later.-,description,-string)
- [fullName](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_profile.htm#:~:text=47.0%20and%20later.-,fullName,-string)
- [layoutAssignments](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_profile.htm#:~:text=for%20a%20call.-,layoutAssignments,-ProfileLayoutAssignments%5B%5D)
- [loginHours](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_profile.htm#:~:text=they%20access%20Salesforce.-,loginHours,-ProfileLoginHours%5B%5D)
- [loginIpRanges](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_profile.htm#:~:text=25.0%20and%20later.-,loginIpRanges,-ProfileLoginIpRange%5B%5D)
- [userLicense](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_profile.htm#:~:text=app%20are%20visible.-,userLicense,-string)

**Reply by sameerseekala1** [reply]
thanks @scolladon

**Reply by sameerseekala1** [reply]
hi @scolladon,
How does this plugin work on permission sets? I added a change to a PS but the delta is creating a PS file with all the changes. Is this expected? Please advice.

**Reply by scolladon** [reply]
Hi @sameerseekala1,

Permission Sets are treated differently by the metadata API.
When a permission set is deployed, its content during the deployment define its content in the org.
It means it is not possible to do partial deployment with permission set.

You can verify that very easily:
1. Create a PSet with FLS on 2 fields on the Account object
2. Retrieve this PSet via sfdx
3. Remove one FLS from the file
4. Deploy the PSet
5. Check in the org, the PSet has now only one FLS configured.

**Reply by sameerseekala1** [reply]
Thanks @scolladon.
Just want to check on a different topic, what is the best way to do backup job using Github Actions? We want to backup our prod into master branch daily so is this possible in Github Actions? Is there any predefined action in Marketplace or a plugin that does the job, please suggest.

**Reply by scolladon** [reply]
I do not have encountered this use case.
What I would do is just use git labels with a date pattern.
I think it is possible to create a schedule workflow with github action to do that

**Reply by sameerseekala1** [reply]
Thanks @scolladon will give a try.
